adaptive/learner/base_learner.py

```diff
@@ -61,30 +61,31 @@ def _wrapped(loss_per_interval):
     return _wrapped
 
 
-class BaseLearner(metaclass=abc.ABCMeta):
+class BaseLearner(metaclass=_RequireAttrsABCMeta):
     """Base class for algorithms for learning a function 'f: X → Y'.
 
     Attributes
     ----------
     function : callable: X → Y
-        The function to learn.
+        The function to learn, the ``learner`` might modify
+        the original function.
```

Comment:
It's not really clear what "the original function" means here. The base class doesn't have an `__init__` or anything

Reply:
I've changed it. Better?